archive/walk.go

```diff
@@ -179,19 +194,38 @@ func (c Chunk) seekIndexPath(ark *Archive) iosrc.URI {
 	return c.tsDir().path(ark).AppendPath(fmt.Sprintf("%s-%s.zng", FileKindSeek, c.Id))
 }
 
+func (c Chunk) metadataPath(ark *Archive) iosrc.URI {
+	return chunkMetadataPath(ark, c.tsDir(), c.Id)
+}
+
 func (c Chunk) Span() nano.Span {
 	return firstLastToSpan(c.First, c.Last)
 }
 
-func (c Chunk) LogID() LogID {
-	name := fmt.Sprintf("%s-%s-%d-%d-%d.zng", c.Kind, c.Id, c.RecordCount, c.First, c.Last)
-	return LogID(path.Join(dataDirname, newTsDir(c.First).name(), name))
+func (c Chunk) RelativePath() string {
+	return path.Join(dataDirname, c.tsDir().name(), fmt.Sprintf("%s-%s.zng", c.Kind, c.Id))
+}
+
+func parseChunkRelativePath(s string) (tsDir, FileKind, ksuid.KSUID, bool) {
+	ss := strings.Split(s, "/")
```

Comment:
You could use filepath functions here to guard against weird stuff, like "//" and "../.." etc. But I'm not sure you would ever expect those strings here.